### PR TITLE
Change disk-type on VM instance

### PR DESCRIPTION
**Pull request by manjilgautam**

### CODE REVIEW

File: tools/gce-change-disktype/README.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+Update disks attached to a GCE instance.
+
+## Script Process
+
+1. Stop the VM instance
+1. Get a list of the disks attached to the VM instance
+1. For each disk attached to the instance
+    1. Detach the disk from the VM instance
+    1. Snapshot the disk
```
Comment (zachberger) [inline tools/gce-change-disktype/README.md:11]: I think there is a step after this that is missing. After snapshotting, a new disk is created.

File: tools/gce-change-disktype/README.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
+# Update GCE Instance disk-type
+
+Update disks attached to a GCE instance.
```
Comment (zachberger) [inline tools/gce-change-disktype/README.md:3]: Script to change disk type of disks attached to a GCE instance.

File: tools/gce-change-disktype/README.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+## Usage
+
+A command-line runnable Python 3 script has been provided and has the 
+following
+usage:
+
+```
+usage: main.py [-h] --project PROJECT --zone ZONE --instance INSTANCE --new_disk_type
+KEYVERSION [--key-global]
```
Comment (zachberger) [inline tools/gce-change-disktype/README.md:25]: There is some leftover stuff from the original script

File: tools/gce-change-disktype/main.py [hunk 4]
```diff
@@ -0,0 +1,277 @@
...
+  operation = compute.disks().delete(
+      project=project, zone=zone, disk=disk).execute()
+  result = wait_for_zonal_operation(compute, project, zone, operation)
+  logging.debug('Deleting disk project=%s, zone=%s, disk=%s complete.', project,
+                zone, disk)
+  return result
+
+def create_disk(compute, project, region, zone, snapshot_name, disk_name,
+                disk_type):
+  """Creates a new CMEK encrypted persistent disk from a snapshot"""
```
Comment (zachberger) [inline tools/gce-change-disktype/main.py:199]: There is some leftover stuff from the original script

File: tools/gce-change-disktype/README.md [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+  --destructive         Upon completion, delete source disks and snapshots
+                        created during migration process.
+```
+
+The script uses Google [Application Default 
+Credentials](https://cloud.google.com/docs/authentication/production).
+
+If further automation is required, e.g. running the process for multiple 
+GCP
+instances, the `migrate_instance_to_cmek` function is a good starting
```
Comment (zachberger) [inline tools/gce-change-disktype/README.md:43]: There is some leftover stuff from the original script

File: tools/gce-change-disktype/main.py [hunk 6]
```diff
@@ -0,0 +1,277 @@
...
+  args = parser.parse_args()
+  create_newdisk_process(args.project, args.zone, args.instance, args.disktype, args.destructive)
+
+def create_newdisk_process(project, zone, instance, disktype, destructive):
+  start = time.time()
+  region = zone.rpartition("-")[0]
+  compute = googleapiclient.discovery.build('compute', 'v1')
+  stop_instance(compute, project, zone, instance)
+  disks = get_instance_disks(compute, project, zone, instance)
+  disktype = 'https://www.googleapis.com/compute/v1/projects/mgtm-sandbox/zones/us-central1-a/diskTypes/' + disktype
```
Comment (zachberger) [inline tools/gce-change-disktype/main.py:79]: Your project is still hardcoded here

File: tools/gce-change-disktype/main.py [hunk 7]
```diff
@@ -0,0 +1,277 @@
...
+  logging.debug('Deleting disk project=%s, zone=%s, disk=%s', project, zone,
+                disk)
+  operation = compute.disks().delete(
+      project=project, zone=zone, disk=disk).execute()
+  result = wait_for_zonal_operation(compute, project, zone, operation)
+  logging.debug('Deleting disk project=%s, zone=%s, disk=%s complete.', project,
+                zone, disk)
+  return result
+
+def create_disk(compute, project, region, zone, snapshot_name, disk_name,
```
Comment (zachberger) [inline tools/gce-change-disktype/main.py:197]: Can this function be deleted? I think there is a new function above.

**Review by zachberger — CHANGES_REQUESTED** [review]
Added some comments, it looks like there are some things lingering from the original code, and there is something hardcoded.

File: tools/gce-change-disktype/README.md (hunk 1 above)
Comment (manjilgautam) [reply tools/gce-change-disktype/README.md:11]: Fixed.

File: tools/gce-change-disktype/README.md (hunk 2 above)
Comment (manjilgautam) [reply tools/gce-change-disktype/README.md:3]: Fixed.

File: tools/gce-change-disktype/main.py (hunk 4 above)
Comment (manjilgautam) [reply tools/gce-change-disktype/main.py:199]: Cleaned it up.

File: tools/gce-change-disktype/README.md (hunk 5 above)
Comment (manjilgautam) [reply tools/gce-change-disktype/README.md:43]: Removed.

File: tools/gce-change-disktype/main.py (hunk 6 above)
Comment (manjilgautam) [reply tools/gce-change-disktype/main.py:79]: Fixed it.

File: tools/gce-change-disktype/main.py (hunk 7 above)
Comment (manjilgautam) [reply tools/gce-change-disktype/main.py:197]: I think we will still need this one.

**Comment by boredabdel** [comment]
@zachberger are you happy with the changes, if yes comment and i will merge it :)

**Comment by zachberger** [comment]
@boredabdel LGTM for merging, thank you!

**Review by zachberger — APPROVED** [review]

**Review by boredabdel — APPROVED** [review]
LGTM

**Pull request merged**